src/features/empty-base-layer/test/emptyBaseLayer.spec.js

```diff
@@ -0,0 +1,53 @@
+describe('ui.grid.emptyBaseLayer', function () {
+
+  var scope, element, timeout, viewportHeight, emptyBaseLayerContainer;
+
+  beforeEach(module('ui.grid.emptyBaseLayer'));
+
+  beforeEach(inject(function (_$compile_, $rootScope, $timeout, $httpBackend) {
+
+    var $compile = _$compile_;
+    scope = $rootScope;
+    timeout = $timeout;
+
+    viewportHeight = "100";
+    scope.gridOptions = {};
+    scope.gridOptions.data = [
+      { col1: 'col1', col2: 'col2' }
+    ];
+    scope.gridOptions.onRegisterApi = function (gridApi) {
+      scope.gridApi = gridApi;
+      scope.grid = gridApi.grid;
+      var renderBodyContainer = scope.grid.renderContainers.body;
+      spyOn(renderBodyContainer, 'getViewportHeight').and.callFake(function() {
+        return viewportHeight;
+      });
+    };
+
+    //$httpBackend.when('GET', 'expandableRowTemplate.html').respond("<div class='test'></div>");
+    element = angular.element('<div class="col-md-5" ui-grid="gridOptions" ui-grid-empty-base-layer></div>');
+
+    $timeout(function () {
+      $compile(element)(scope);
+    });
+    $timeout.flush();
```

Comment:
Why are you wrapping the compilation in a timeout? You could just as easily add a scope.$apply() function call after compiling the element instead of adding a timeout.